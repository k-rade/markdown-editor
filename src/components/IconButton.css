.icon-button {
  display: flex;
  align-items: center;
  justify-content: center;
  outline: none;
  border: none;
  background-color: transparent;
  font-weight: 500;
  font-size: 1rem;
  color: var(--color-text);
  cursor: pointer;
  border-radius: 50%;
  padding: 0.625rem;
  position: relative;
}

.icon-button:hover {
  background-color: var(--color-gray-300);
}

.icon-button:hover svg>path {
  stroke: var(--color-white-two);
}

.icon-button:disabled {
  cursor: not-allowed;
}

.icon-button-small .icon {
  width: 1rem;
  height: 1rem;
}