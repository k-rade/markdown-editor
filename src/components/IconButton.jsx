import "./IconButton.css";

const IconButton = ({ icon, onClick, tooltip, size = "medium", ...props }) => {
  return (
    <button
      {...props}
      className={`icon-button icon-button-${size}`}
      type="button"
      onClick={onClick}
    >
      {icon}
      {tooltip && <div className="icon-button-tooltip">{tooltip}</div>}
    </button>
  );
};

export default IconButton;
