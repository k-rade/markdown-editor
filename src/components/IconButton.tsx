import "./IconButton.css";

interface Props {
  icon: JSX.Element;
  onClick?: () => void;
  size?: "small" | "medium";
  disabled?: boolean;
}

const IconButton = ({
  icon,
  onClick,
  size = "medium",
  disabled,
}: Props): JSX.Element => {
  return (
    <button
      disabled={disabled}
      className={`icon-button icon-button-${size}`}
      type="button"
      onClick={onClick}
    >
      {icon}
    </button>
  );
};

export default IconButton;
