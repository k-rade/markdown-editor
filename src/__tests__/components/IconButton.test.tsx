import { render, fireEvent } from "@testing-library/react";
import IconButton from "../../components/IconButton";

describe("IconButton", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <IconButton icon={<>icon</>} tooltip="tooltip text" />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should call onClick", (): void => {
    const mockOnClick = jest.fn();
    const { getByText } = render(
      <IconButton icon={<>icon</>} onClick={mockOnClick} />
    );

    fireEvent.click(getByText("icon"));

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
