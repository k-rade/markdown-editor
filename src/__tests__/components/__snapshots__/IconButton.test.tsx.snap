// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render 1`] = `
<body>
  <div>
    <button
      class="icon-button icon-button-medium"
      type="button"
    >
      icon
    </button>
  </div>
</body>
`;
